gravity.sh

```diff
@@ -591,8 +592,40 @@ gravity_DownloadBlocklistFromUrl() {
     fi
   fi
 
-  # shellcheck disable=SC2086
-  httpCode=$(curl --connect-timeout ${curl_connect_timeout} -s -L ${compression} ${cmd_ext} ${heisenbergCompensator} -w "%{http_code}" "${url}" -o "${listCurlBuffer}" 2>/dev/null)
+  # If we are going to "download" a local file, we first check if the target
+  # file has a+r permission. We explicitly check for all+read because we want
+  # to make sure that the file is readable by everyone and not just the user
+  # running the script.
+  if [[ $url == "file://"* ]]; then
+    # Get the file path
+    file_path=$(echo "$url" | cut -d'/' -f3-)
+    # Check if the file exists and is a regular file (or a symlink to one)
+    if [[ ! -e $file_path ]]; then
+      # Output that the file does not exist
+      echo -e "${OVER}  ${CROSS} ${file_path} does not exist"
+      download=false
+    elif [[ ! -f $file_path ]]; then
+      # Output that the file is not a regular file
+      echo -e "${OVER}  ${CROSS} ${file_path} is not a regular file"
+      download=false
+    else
+      # Check if the file has a+r permissions
+      permissions=$(stat -Lc "%a" "$file_path")
```

Comment:
This is more of a style request but can we use the full name of the flags when we call them in scripts? Saves having to track down the man page to know what it's doing.

```suggestion
      permissions=$(stat --dereference --format="%a" "$file_path")
```